### PR TITLE
Payment processing with view user files api

### PR DESCRIPTION
**Pull request by taha-transdata**
### **Description**

-- Add new Migrations
-- Add new trait for stripe payment processing
-- Process the payment to strip
-- My Files Api for client

### CODE REVIEW

File: app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -67,4 +90,44 @@ public function destroy(string $id)
...
+                return setErrorResponse("Validation error's", $validator->errors()->messages());
+            }
+            $state=State::where('id',$request->state_id)->first();
+            if($state!=null){
+                $userStateData = [
+                    'user_id'=>$request->user_id,
+                    'state_id'=>$request->state_id,
+                    'price'=>$request->price,
+                ];
+                if($state->price>=$request->price){
```
Comment (mudassarse) [inline app/Http/Controllers/UserController.php:114]: We can enter any price for a state, not only a larger price. Additionally, this should be handled through a relationship. Users with states should have a 'belongsToMany' relationship. Use a pivot table for that purpose

Comment (taha-transdata) [reply app/Http/Controllers/UserController.php:114]: @ if($state->price>=$request->price) this condition is already checking the user state price should be lesser from original price.

File: app/Http/Controllers/StateController.php [hunk 2]
```diff
@@ -12,7 +14,16 @@ class StateController extends Controller
...
+        try {
+            $states = State::all();
+            if (!$states->isEmpty()) {
+                return setSuccessResponse("States with price retrieved successfully", StateResource::collection($states));
+            } else {
+                return setErrorResponse('States with price not found', []);
+            }
+        } catch (\Exception $e) {
+            return setErrorResponse('Something went wrong on server !!', []);
+        }
```
Comment (mudassarse) [inline app/Http/Controllers/StateController.php:26]: Please add save log in every catch case.

File: app/Http/Controllers/UserController.php [hunk 3]
```diff
@@ -67,4 +91,46 @@ public function destroy(string $id)
...
+                'user_id' => 'required',
+                'state_id' => 'required',
+                'price' => 'required|decimal:2'
+            ]);
+            if ($validator->fails()) {
+                return setErrorResponse("Validation error's", $validator->errors()->messages());
+            }
+            $state=State::find($request->state_id);
+            $user=User::find($request->user_id);
+            if($state!=null && $user!=null ){
```
Comment (mudassarse) [inline app/Http/Controllers/UserController.php:110]: use is_null() method to check null object. @taha-transdata

File: app/Http/Controllers/UserController.php (hunk 1 above)
Comment (mudassarse) [reply app/Http/Controllers/UserController.php:116]: @taha-transdata  Why are putting condition then?

**Pull request merged**